### DISCUSSION TITLE
How to change file in locales?

### DISCUSSION THREAD

**Discussion by SviridoffA** [apache/cloudstack#9592] [Q&A] [answered]
Hi! Please I need help! Is it possible, to make changes in locales? I'm trying to make some changes in /usr/share/cloudstack-management/webapp/locales/, in json file. For example, trying to add login.label.portal which is absent for my locale. I'm editing json and restarting management server, but changes not implementing. Is there any right way to do this?

**Comment by SviridoffA** [comment]
Looks like it's not working like this. To implement this change I need make changes in code and use npm to create a new static web application. I have tryed but unfortunaly dont have enough skills for this.

**Reply by DaanHoogland** [reply]
@SviridoffA , if your change is generic you can add a PR for your locale to the github repo. Hopefully we can find someone to check if you are proficient in your language ;)

**Reply by shwstppr** [reply, marked as answer]
> Looks like it's not working like this. To implement this change I need make changes in code and use npm to create a new static web application. I have tryed but unfortunaly dont have enough skills for this. 

@SviridoffA Editing locale JSON file in your installation should work fine. If it is not showing then make sure it is not a caching issue. Try an incognito window after restarting management server

**Reply by SviridoffA** [reply]
@DaanHoogland , if our final choice would be Cloudstack I will make it)) Its not too scary for us if some words translated not perfect, but its critical when some keys in json are absent. like this:

![2024-08-27_17-30](https://github.com/user-attachments/assets/b805b48d-3e94-4c6d-9734-b5adcf79d593)

**Reply by SviridoffA** [reply]
@shwstppr Thx!!!  It was realy just a cache issue. I have try update page without cache, but its not work for some reason, but in incognito all changes are showing

![2024-08-27_17-36](https://github.com/user-attachments/assets/953c5dd8-655a-410c-961e-37c4c534d1bc)

Its looks like icon is gone some where. I think its related with missed keys too..   I will try to figure out this.
